### PR TITLE
Add tensorboard tracking if log_on_tensorboard is True + add GDL loss

### PR DESCRIPTION
**Pull request by albansteff**
Add `"log_on_tensorboard": true` in file `train.json` to log losses and metrics on tensorboard.

Add Generalized Dice Loss as a loss function for training.

### CODE REVIEW

File: segmentation/run_model.py [hunk 1]
```diff
@@ -435,6 +451,9 @@ def train_loop(self, save_model=True):
                     self.model.eval()
                     validation_loss = self.train_loop()
                     self.model.train()
+                    self.tb_logger.add_scalars('Average Train/Val loss',
+                                               {'Train': average_loss,
+                                                'Val': validation_loss}, self.epoch)
```
Comment (romainVala) [inline segmentation/run_model.py:457]: il manque un 
if self.log_on_tensorboard is True:

**Review by romainVala — APPROVED** [review]
